### PR TITLE
Alpha preview

### PR DESCRIPTION
**Pull request by chebizarro**
### *This is a preview only and should not be used in a production environment!*

This PR brings the repo to am Android buildable state where it can interact with a Mostro daemon and Nostr relay.

On startup, it connects to a relay running on port 7000 on either 127.0.0.1 or 10.0.2.2 if running in an emulator. The relays and the mostro daemon public key can be configured in the file `lib/core/config.dart`

Currently this preview allows for adding Buy/Sell Orders

It still lacks a UI for finalizing orders

Only ephemeral keys are used for orders, so once again, *do not use in a production environment* 

## Release Notes

- **New Features**
	- Introduced new classes for handling orders, payments, and user sessions, enhancing order management capabilities.
	- Added support for currency selection and exchange rate display through new widgets.
	- Implemented localization support for various user interactions within the app.

- **Bug Fixes**
	- Removed deprecated secure storage functionalities, ensuring a cleaner codebase.

- **Refactor**
	- Transitioned to Riverpod for state management, improving the overall architecture and responsiveness of the app.
	- Updated event handling and order processing methods to streamline functionality.

- **Documentation**
	- Enhanced documentation for new classes and methods, providing clearer guidance for developers.

### CODE REVIEW

**Review by Catrya — CHANGES_REQUESTED** [review]
Hi @chebizarro great job! congratulations.

There are some things that can be improved when displaying orders such as:

- Orders that have ranges are not being displayed correctly, it only shows the lower value. For example: purchase order of 210-600 cup, it only shows one order for 210 cup
diff


- If the maker of the order has `"rating", "none"` the client displays the order, but if the maker was previously rated, this appears in the logs `Another exception was thrown: FormatException: Missing rating object in JSON` and the client shows a red error. Also, it does not show any order that has been published after that one 
Captura desde 2024-11-21 11-33-26

**Review by Catrya — COMMENTED** [review]
I think that when someone presses an order to see its details, if they don't want to take it, and they press Cancel, there is no need to display the message that you have cancelled the order, because in reality the user didn't cancel it, they just didn't take it.

I think that automatically returning the user to the main screen would be fine, without them having to press another button to do so.

Captura desde 2024-11-21 12-21-47

**Review by Catrya — COMMENTED** [review]
In the purchase orders, it says: `offering X sats for Y fiat` but it seems incorrect to me that way, since the buyer offers the fiat, not the btc. 

I think it should swap the order in which they appear and put `offering Y fiat for X sats`, and if it is a market price order perhaps it should say `offering Y fiat for sats` without the value.

Another way to put it would be (using the fiat USD example): 
For purchase orders:
- Market price orders: `Buying Sats for X USD`
- Fixed orders:
`Buying Y sats for X USD`

For sales orders:
- Market price orders:
`Selling sats for X USD`
- Fixed orders:
`Selling Y sats for X USD`

What do you think @chebizarro ?

**Comment by Catrya** [comment]
@chebizarro This might be useful for UI https://github.com/MostroP2P/mobile/discussions/26
If you want we can talk about more ideas in that UI discussion.

**Comment by Catrya** [comment]
>  Currently this preview allows for:
> - Take Buy/Sell Orders

Hi @chebizarro Sorry for so many messages 😅 I can't get it to take an order. When I press Continue to take an order, the order remains in pending status. It seems that Mostro is not receiving the message from takesell or takebuy. 
Maybe I'm doing something wrong when testing it. Any suggestions on how to do this?

File: lib/core/utils/nostr_utils.dart
Comment (chebizarro) [inline lib/core/utils/nostr_utils.dart:151]: Support for encrypting DMs to send to the mostro daemon as per issue #3

File: lib/core/utils/nostr_utils.dart
Comment (chebizarro) [inline lib/core/utils/nostr_utils.dart:216]: Support for decrypting DMs from mostro as per issue #3

File: lib/data/models/enums/action.dart
Comment (chebizarro) [inline lib/data/models/enums/action.dart:1]: Introduces the Actin type for the mostro message as per issue #3

File: lib/data/models/mostro_message.dart
Comment (chebizarro) [inline lib/data/models/mostro_message.dart:1]: Implements the mostro message kind as per #3

File: pubspec.yaml
Comment (chebizarro) [inline pubspec.yaml:54]: NIP-44 support was implemented as per issue #3 but bundled into a separate library as it has utility for other Nostr developers. This library can be transferred to the mostro project if desired

File: lib/data/models/session.dart
Comment (chebizarro) [inline lib/data/models/session.dart:1]: The Session object addresses issue #28

**Review by Catrya — APPROVED** [review]
Hi @chebizarro, thanks for considering the comments I made to you. I've already test it and it shows the pending orders very well

Take sell/buy orders can be left for another PR, no problem, great job!

**Comment by grunch** [comment]
> The latest commits address:
> 
>     * @grunch request to implement Code Rabbit AI suggestions
> 
>     * @Catrya bugs/suggestions with the exception of Take/Sell Orders
>       After this PR is merged I will be able to make much smaller PRs that will address specific issues

Understood, but the description of this PR has this written:

```
Currently this preview allows for:

    Add Buy/Sell Orders
    Take Buy/Sell Orders
```

Could you please update this removing what we are not going to have on this PR so we can merge it? thanks

**Comment by grunch** [comment]
Hi @chebizarro this is ready to be merged?

**Comment by chebizarro** [comment]
> Hi @chebizarro this is ready to be merged?

It is, I edited the original the original PR comment to reflect the actual features included. Once this is merged, I'll open another PR with the complete take buy/sell and make buy/sell order features from the development branch.

**Comment by grunch** [comment]
great! please next time ping me so we can merge right after is done

**Review by grunch — APPROVED** [review]
tACK 

great job @chebizarro !!!

**Pull request merged**
